### DISCUSSION TITLE
how to implement a grap with grid

### DISCUSSION THREAD

**Discussion by GazeboXu** [eclipse-glsp/glsp#1368] [General] [answered]
hello
  If I want to implment a graph with grid, what's the best way to implement it? Is there a IView already has such a feature?
thanks.

**Comment by tortmayr** [comment, marked as answer]
Hi @GazeboXu,
we recently implemented a feature to support this. It will be part of the upcoming 2.2.x release (targeted for the end of this week)
With this new feature the default graph views `GGraphView` and `GLSPProjectionView` support grid rendering out of the box.
The behavior is deactivated by default to enable it you have to configure the newly introduced `gridModule` for you diagram language.
- Grid styling can be controlled via css
- Grid enablement can be controlled via action and/or the tool palette
Documentation for this will be added once the 2.2.x release is completed.
Here is a quick demo of  the grid feature:

https://github.com/user-attachments/assets/f344c02a-2fce-4684-aab0-5a9f8f74feed

**Comment by GazeboXu** [comment]
got it, thanks.
